### DISCUSSION TITLE
Processes / Workflows

### DISCUSSION THREAD

**Discussion by keyseven123** [nebulastream/nebulastream#6] [Ideas]
Here, we should discuss/document our processes and workflows.

**Comment by kykrueger** [comment]
@keyseven123 is this intended to cover devops processes and workflows?
I'm working on code-coverage in the old nebulastream repo at the moment, and would like to make sure the commits from that PR get cherry-picked or merged into this new repository when they are ready.

**Reply by keyseven123** [reply]
Not really for DevOps processes but for general processes.

**Reply by kykrueger** [reply]
> Not really for DevOps processes but for general processes.

Ok, maybe we could put these into some categories, I suppose the following would belong:
 - permissions concept, which permissions should be managed in the roles
 - nebula-stream project role concept (grouped into github teams)
     - process for adding members
     - process for removing members
 - Design decisions / technical ticket planning

I could think of more requirements similar to these, but am not sure if this is now what you have in mind for general processes.

**Comment by zeuchste** [comment]
Hi all maybe we should start here first a list of process and workflows. Maybe it would be good to categorize them like
- Membership
- Onboarding
- Code Change
- Release 
- Code Contribution
- DevOps
- Bug handling
- Etc

**Comment by keyseven123** [comment]
### Github Discussions --> Issues
I had some discussions with @adrianmichalke  and @alepping  about going from github discussion to issues for our milestone and sprint respectively. We came to the following result.

#### Before the milestone meeting:
Every group/component creates an EPIC issue for each of the things that should be part of the upcoming milestone. 
Each EPIC issues should contain all information necessary to implement it. This means that there should exist a design document. This design document can be written directly in the epic issue or if as a standalone document under `nebulastream-public/docs`. 
Additionally, each EPIC issue should have a time estimate how long it should take at least 2 sprint cycles (so 4 weeks).  The maintainers can overrule/have a veto regarding the prioritization. 

#### During the milestone meeting:
As a group, we discuss what EPICs to prioritize and what to put into our backlog for doing them in a next milestone. For all of the EPIC issues that we plan on doing the upcoming milestone, we discuss the time estimate and who should be responsible for it. At least one maintainer has to be part of every EPIC.


#### During the milestone:
Before each sprint, people create issues belonging to the EPICs they were assigned to. These issues should reference the EPIC and also contain all of the necessary information to implement the issue.
